src/odemis/driver/xt_toolkit_client.py

```diff
@@ -0,0 +1,233 @@
+            unit=beam_stigmator_info["unit"],
+            range=beam_stigmator_range,
+            setter=self._setBeamStigmator)
+
+        pattern_stigmator_info = self.parent.secondary_stigmator_info()
+        pattern_stigmator_range_x = pattern_stigmator_info["range"]["x"]
+        pattern_stigmator_range_y = pattern_stigmator_info["range"]["y"]
+        pattern_stigmator_range = tuple((i, j) for i, j in zip(pattern_stigmator_range_x,
+                                                                   pattern_stigmator_range_y))
+        self.patternStigmator = model.TupleContinuous(
+            tuple(self.parent.get_secondary_stigmator()),
+            unit=pattern_stigmator_info["unit"],
+            range=pattern_stigmator_range,
+            setter=self._setPatternStigmator)
+
+        self.beamShiftTransformationMatrix = model.ListVA(
+            self.parent.get_dc_coils(),
+            unit=None,
+            readonly=True)
+
+        self.multiprobeRotation = model.FloatVA(
+            math.radians(self.parent.get_mpp_orientation()),
+            unit="rad",
+            readonly=True)
+
+        aperture_index_info = self.parent.aperture_index_info()
+        self.apertureIndex = model.IntContinuous(
+            int(self.parent.get_aperture_index()),
+            unit=aperture_index_info["unit"],
+            range=tuple(int(i) for i in aperture_index_info["range"]),
+            setter=self._setApertureIndex)
+
+        beamlet_index_info = self.parent.beamlet_index_info()
+        beamlet_index_range_x = beamlet_index_info["range"]["x"]
+        beamlet_index_range_y = beamlet_index_info["range"]["y"]
+        beamlet_index_range = tuple((int(i), int(j)) for i, j in zip(beamlet_index_range_x, beamlet_index_range_y))
+        beamlet_index = self.parent.get_beamlet_index()
+        self.beamletIndex = model.TupleContinuous(
+            tuple(int(i) for i in beamlet_index),  # convert tuple values to integers.,
+            unit=beamlet_index_info["unit"],
+            range=beamlet_index_range,
+            setter=self._setBeamletIndex
+        )
+
+        multibeam_mode = (self.parent.get_use_case() == 'MultiBeamTile')
+        self.multiBeamMode = model.BooleanVA(
+                multibeam_mode,
+                setter=self._setMultiBeamMode
+        )
+
+    def _updateSettings(self):
+        """
+        Read all the current settings from the SEM and reflects them on the VAs
+        """
+        # TODO When the new approach of adding update function to a list is implemented in xt_client.py instead of
+        #  overwriting the method updateSettings the method _updateMBSettings can be added to the list of
+        #  functions to be updated.
+
+        # Polling XT client settings
+        super(Scanner, self)._updateSettings()
+        # Polling XTtoolkit settings
+        # TODO K.K. how to remove waiting but not have error because of inheritance.
+        import time
+        time.sleep(1)
```

Comment:
Why do you need a sleep() here? The code should be executed one after the other, and all be fine, no?

Reply:
This is an error (witht he TODO) I should have fixed. I think I found a better solution than the sleep but I forgot to implement it.
Because of the update thread in the super the polling cannot start directly, because the multibeam part isn't initialized by then. I should change something in the __init__ of the scanner with the order of the VA's and the super.